src/collections.ts

```diff
@@ -184,9 +184,49 @@ export class Py_List extends Py_Object implements Iterable {
     }
     return None;
   }
+
   public __delitem__(key: IPy_Object): IPy_Object {
     // Delete is the same as splicing out the element and moving everything down
-    this._list.splice(standardizeKey(key, this._list.length), 1);
+    if (key.getType() === enums.Py_Type.SLICE){
+      var slice = <Py_Slice> key,
+      step = slice.step === None ? 1 : (<Py_Int | Py_Long> slice.step).toNumber();
+      if (step === 1){
+        var start = slice.start === None ? 0 : (<Py_Int> slice.start).toNumber(),
+          stop = slice.stop === None ? this._list.length : (<Py_Int> slice.stop).toNumber(),
+          len = stop - start < 0 ? 0 : stop - start;
+        Array.prototype.splice.apply(this._list, [start, len]);
```

Comment:
Oh ya, in this case I should have used the latter form. In other places, I am using apply because splice takes variable number of arguments for the elements to be added instead of an Array.